### PR TITLE
Sean schema

### PR DESCRIPTION
**Pull request by SeanEZhou**
- Implemented Schemas for Order, Drone and Order status.
- Implemented Validation tests for above schemas.

### CODE REVIEW

File: test/order_status_schema.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        m.validate(function(err) {
+            expect(err.errors.orderId).to.exist;
+            done();
+        });
+    });
+
+    it('order status should always contain a customer id', function(done) {
+        m = new OrderStatus();
+        m.validate(function(err) {
+            expect(err.errors.droneId).to.exist;
```
Comment (phi-h-nguyen) [inline test/order_status_schema.test.js:16]: I think this should be customerId

**Review by phi-h-nguyen — CHANGES_REQUESTED** [review]
I haven't worked with mongoose before, but it mostly looks good! I just made one comment on the order_status_schema.test about the customerId attribute, but other than that lgtm!

**Comment by McSobkov** [comment]
I think based on the question you linked in slack and given that Phi is matching the pattern, you may want to switch to this structure

image

I also moved my tests to a subfolders of test that I'm calling 'schema'. I don't know how many types of tests we will be writing but we may end up with 20 test files and it might be better to keep them in 4 different folders.

If we're switching to mocha/chai, that's fine, but it should be reflected in the README (I'm not sure if the two are related or not, but it seems like you're only using chai here)?

I imagine we'll do more testing in the next couple of days, but we'll need some way of doing unique IDs since mongoose seems to just use the default mongo id's which are pretty ugly. That's more just to keep it on your radar than an update for this PR!

**Review by McSobkov — CHANGES_REQUESTED** [review]
I think based on the question you linked in slack and given that Phi is matching the pattern, you may want to switch to this structure:

image

I also moved my tests to a subfolders of test that I'm calling 'schema'. I don't know how many types of tests we will be writing but we may end up with 20 test files and it might be better to keep them in 4 different folders.

If we're switching to mocha/chai, that's fine, but it should be reflected in the README (I'm not sure if the two are related or not, but it seems like you're only using chai here)?

I imagine we'll do more testing in the next couple of days, but we'll need some way of doing unique IDs since mongoose seems to just use the default mongo id's which are pretty ugly. That's more just to keep it on your radar than an update for this PR!

File: test/order_status_schema.test.js (hunk 1 above)
Comment (SeanEZhou) [reply test/order_status_schema.test.js]: yeah good catch, the description should be “always contain a drone id"

**Comment by SeanEZhou** [comment]
Thanks for the feedback guys! I’ve fixed the issues:

- refactored backend into its own directory
- fixed error in test description
- updated README 

Please review my changes so that we can merge this branch in. Thanks!

**Review by McSobkov — APPROVED** [review]

**Pull request merged**